test/framework/config.py

```diff
@@ -300,8 +301,10 @@ def test_generaloption_config_file(self):
             '/tmp/foo',
             os.path.join(os.getenv('HOME'), '.local', 'easybuild', 'ebfiles_repo'),
             os.path.join(tmpdir, 'easybuild', 'easyconfigs'),
+            os.getenv('HOME'),
+            os.path.join('/tmp',os.getenv('USER')),
         ]
-        self.assertEqual(options.robot_paths[:3], robot_paths)
+        self.assertEqual(options.robot_paths[:5], robot_paths)
```

Comment:
turn out the error is pretty confusing when the comparison is done like this (as in, I couldn't figure it out and had to break this up into separate tests), so let's change this to:

```python
# hardcoded first entry
self.assertEqual(options.robot_paths[0], '/tmp/foo')
# resolved value for %(sourcepath)s template
self.assertEqual(options.robot_paths[1], os.path.join(os.getenv('HOME'), '.local', 'easybuild', 'ebfiles_repo'))
# resolved value for HOME constant
self.assertEqual(options.robot_paths[2], os.getenv('HOME'))
# resolved value that uses USER constant
self.assertEqual(options.robot_paths[3], os.path.join('/tmp', os.getenv('USER')))
# first path in DEFAULT_ROBOT_PATHS
self.assertEqual(options.robot_paths[4], os.path.join(tmpdir, 'easybuild', 'easyconfigs'))
```